src/Raven.Server/Web/Studio/DataDirectoryInfo.cs

```diff
@@ -0,0 +1,216 @@
+            {
+                context.Write(writer, dataDirectoryResult.ToJson());
+            }
+        }
+
+        private List<string> GetRelevantNodes(string databaseName, ClusterTopology clusterTopology)
+        {
+            if (databaseName == null)
+                return clusterTopology.AllNodes.Keys.ToList();
+
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var rawDatabaseRecord = _serverStore.Cluster.ReadRawDatabase(context, databaseName, out _);
+                if (rawDatabaseRecord == null)
+                    return new List<string>();
+
+                var databaseTopology = _serverStore.Cluster.ReadDatabaseTopology(rawDatabaseRecord);
+                if (databaseTopology == null)
+                    return new List<string>();
+
+                return databaseTopology.AllNodes.ToList();
+            }
+        }
+
+        private async Task UpdateNodesDirectoryResult(
+            IEnumerable<string> nodes, ClusterTopology clusterTopology,
+            string urlPath, DataDirectoryResult dataDirectoryResult)
+        {
+            var httpClient = RequestExecutor.CreateHttpClient(_serverStore.Server.Certificate.Certificate, DocumentConventions.Default);
+            var tasks = new List<Task<SingleNodeDataDirectoryResult>>();
+
+            foreach (var nodeTag in nodes)
+            {
+                _serverStore.ServerShutdown.ThrowIfCancellationRequested();
+
+                if (nodeTag.Equals(_serverStore.NodeTag, StringComparison.OrdinalIgnoreCase))
+                    continue;
+
+                var serverUrl = clusterTopology.GetUrlFromTag(nodeTag);
+                if (serverUrl == null)
+                    continue;
+
+                tasks.Add(Task.Run(async () =>
+                {
+                    var url = $"{serverUrl}/{urlPath}";
+                    var singleNodeResult = await GetSingleNodeDataDirectoryInfo(url, httpClient);
+                    return singleNodeResult;
+                }, _serverStore.ServerShutdown));
+            }
+
+            await Task.WhenAll(tasks);
+
+            foreach (var task in tasks)
+            {
+                if (task.IsCompletedSuccessfully == false)
+                    continue;
+
+                var singleNodeResult = await task;
+                if (singleNodeResult == null)
+                    continue;
+
+                dataDirectoryResult.List.Add(singleNodeResult);
+            }
+        }
+
+        private async Task<SingleNodeDataDirectoryResult> GetSingleNodeDataDirectoryInfo(string url, HttpClient httpClient)
+        {
+            using (var cts = new CancellationTokenSource(7 * 1000))
+            {
+                try
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, url);
```

Comment:
We should use custom RavenCommand instead of doing all of this? We have a RequestExecutor here available.

Reply:
Then you would not need any HttpClient here and all of those changes made there.

Reply:
Agree, we don't want to manually use these

Reply:
done